reVX/hybrid_stats/hybrid_stats.py

```diff
@@ -0,0 +1,829 @@
+# -*- coding: utf-8 -*-
+"""
+Module to compute hybrid solar-wind generation correlations
+"""
+from concurrent.futures import as_completed
+import logging
+import numpy as np
+import os
+import pandas as pd
+from scipy.spatial import cKDTree
+from scipy.stats import pearsonr, spearmanr, kendalltau
+
+from rex.resource import Resource
+from rex.utilities.execution import SpawnProcessPool
+from rex.utilities.utilities import get_lat_lon_cols
+
+logger = logging.getLogger(__name__)
+
+
+class HybridStats:
+    """
+    Compute comparitive correlations for co-located wind and solar generation
+    """
+    def __init__(self, solar_h5, wind_h5, statistics='pearson',
+                 res_cls=Resource, hsds=False):
+        """
+        Parameters
+        ----------
+        solar_h5 : str
+            Path to solar h5 file(s)
+        wind_h5 : str
+            Path to wind h5 file(s)
+        statistics : str | tuple | dict, optional
+            Statistics to extract, either a key or tuple of keys in
+            cls.STATS, or a dictionary of the form
+            {'stat_name': {'func': *, 'kwargs: {**}}},
+            by default 'pearson'
+        res_cls : Class, optional
+            Resource class to use to access res_h5, by default Resource
+        hsds : bool, optional
+            Boolean flag to use h5pyd to handle .h5 'files' hosted on AWS
+            behind HSDS, by default False
+        """
+        self._solar_h5 = solar_h5
+        self._wind_h5 = wind_h5
+        self._stats = None
+        self.statistics = statistics
+
+        self._res_cls = res_cls
+        self._hsds = hsds
+
+        self._meta, self._time_index, self._solar_ts, self._wind_ts = \
+            self._pre_flight_check()
+
+    @property
+    def solar_h5(self):
+        """
+        Path to solar h5 file(s)
+
+        Returns
+        -------
+        str
+        """
+        return self._solar_h5
+
+    @property
+    def wind_h5(self):
+        """
+        Path to wind h5 file(s)
+
+        Returns
+        -------
+        str
+        """
+        return self._wind_h5
+
+    @property
+    def statistics(self):
+        """
+        Dictionary of statistic functions/kwargs to run
+
+        Returns
+        -------
+        dict
+        """
+        return self._stats
+
+    @statistics.setter
+    def statistics(self, statistics):
+        """
+         Statistics to extract, either a key or tuple of keys in
+        cls.STATS, or a dictionary of the form
+        {'stat_name': {'func': *, 'kwargs: {**}}}
+
+        Parameters
+        ----------
+        statistics : dict
+        """
+        self._stats = self._check_stats(statistics)
```

Comment:
I feel like self._check_stats() could just be put in this property setter since its the only place its called.

Reply:
Good call, will do